lib/mix/lib/mix/tasks/compile.ex

```diff
@@ -85,6 +85,9 @@ defmodule Mix.Tasks.Compile do
   def run(args) do
     Mix.Project.get!()
     Mix.Task.run("loadpaths", args)
+    {opts, _, _} = OptionParser.parse(args, switches: [erl_config: :string])
+
+    load_mix_config(args)
```

Comment:
`persist_erl_config` or `load_erl_config` would be better names. We also need to document the flag.